src/features/organizations/utils/generateTreeData.spec.ts

```diff
@@ -0,0 +1,219 @@
+    const organizations: ZetkinOrganization[] = [
+      mockOrganization({
+        id: 1,
+        parent: null,
+        title: 'Party A',
+      }),
+      mockOrganization({
+        id: 2,
+        parent: null,
+        title: 'Party B',
+      }),
+    ];
+
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: { id: 3, parent: null, title: 'Party C' },
+        role: 'organizer',
+      }),
+      mockMembership({
+        organization: { id: 4, parent: null, title: 'Party D' },
+        role: 'admin',
+      }),
+      mockMembership({
+        organization: { id: 5, parent: null, title: 'Party E' },
+        role: null,
+      }),
+    ];
+
+    const result = generateTreeData(organizations, memberships);
+
+    const expectedTreeData: TreeItemData[] = [];
+
+    expect(result).toEqual(expectedTreeData);
+  });
+
+  it('considers a child org a top org if the user doesnt have access to parent org', () => {
+    const organizations: ZetkinOrganization[] = [
+      mockOrganization({
+        id: 1,
+        parent: null,
+        title: 'Party A',
+      }),
+      mockOrganization({
+        id: 2,
+        parent: {
+          id: 1,
+          title: 'Party A',
+        },
+        title: 'Party B',
+      }),
+    ];
+
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: {
+          id: 2,
+          parent: {
+            id: 1,
+            title: 'Party A',
+          },
+          title: 'Party B',
+        },
+        role: 'organizer',
+      }),
+    ];
+
+    const result = generateTreeData(organizations, memberships);
+
+    const expectedTreeData: TreeItemData[] = [
+      {
+        avatar_file: null,
+        children: [],
+        country: 'SE',
+        email: null,
+        id: 2,
+        is_active: false,
+        is_open: false,
+        is_public: true,
+        lang: null,
+        parent: null,
+        phone: null,
+        slug: 'slug',
+        title: 'Party B',
+      },
+    ];
+
+    expect(result).toEqual(expectedTreeData);
+  });
+
+  it('handles a missing organization without crashing', () => {
+    const organizations: ZetkinOrganization[] = [];
+    const memberships: ZetkinMembership[] = [
+      mockMembership({
+        organization: { id: 1, parent: null, title: 'Party A' },
+        role: 'admin',
+      }),
+    ];
+
+    const expectedTreeData: TreeItemData[] = [];
+
+    const treeData = generateTreeData(organizations, memberships);
+
+    expect(treeData).toEqual(expectedTreeData);
+  });
+});
+
+function mockOrganization(
+  organization: Pick<ZetkinOrganization, 'id' | 'title' | 'parent'>
+): ZetkinOrganization {
+  const injectedProperties = {
+    avatar_file: null,
+    country: 'SE',
+    email: null,
+    is_active: false,
+    is_open: false,
+    is_public: true,
+    lang: null,
+    parent: null,
+    phone: null,
+    slug: 'slug',
+  };
+
+  return { ...organization, ...injectedProperties };
```

Comment:
On this line, whatever is in `injectedProperties` overrides whatever is in `organization`. But `organization` is the data that the user of the function (the tests) are passing in as the definition of the organization.

This means, that all organizations returned by this function will always have `parent: null`, because that's what the `injectedProperties` object has.

This lead to false positives, i.e. tests were passing not because the algorithm they were testing was correct, but because the tests themselves were wrong. That's always a risk when writing tests, especially when writing them _after_ the code under test, because when you write the tests first, before you implement the logic ([TDD](https://en.wikipedia.org/wiki/Test-driven_development)), you will notice if they're not failing at first.